src/yarp.c

```diff
@@ -2224,8 +2224,22 @@ parse_parameters(yp_parser_t *parser) {
         local.end -= 1;
         yp_token_list_append(&parser->current_scope->as.scope.locals, &local);
 
-        yp_node_t *param = yp_node_keyword_parameter_node_create(parser, &name);
-        yp_node_list_append(parser, params, &params->as.parameters_node.keywords, param);
+        if (parser->current.type != YP_TOKEN_COMMA && parser->current.type != YP_TOKEN_PARENTHESIS_RIGHT) {
+          yp_node_t *value = parse_expression(parser, BINDING_POWER_NONE, "Expected to find a default value for the keyword parameter.");
+          yp_node_t *param = yp_node_keyword_parameter_node_create(parser, &name, value);
+          yp_node_list_append(parser, params, &params->as.parameters_node.keywords, param);
+
+          // If parsing the value of the keyword parameter resulted in error recovery,
+          // then we can put a missing node in its place and stop parsing the
+          // parameters entirely now.
+          if (parser->recovering) return params;
+
+          parsing = false;
+        } else {
+          yp_node_t *param = yp_node_keyword_parameter_node_create(parser, &name, NULL);
+          yp_node_list_append(parser, params, &params->as.parameters_node.keywords, param);
+        }
```

Comment:
There is opportunity to refactor the duplicate `yp_node_keyword_parameter_node_create` calls but for now I just wanted to confirm if the new direction made sense.